InvenTree/label/api.py

```diff
@@ -136,17 +138,47 @@ def filter_queryset(self, queryset):
 class LabelPrintMixin(LabelFilterMixin):
     """Mixin for printing labels."""
 
+    rolemap = {
+        "GET": "view",
+        "POST": "view",
+    }
+
+    def check_permissions(self, request):
+        """Override request method to GET so that also non superusers can print using a post request."""
+        if request.method == "POST":
+            request = clone_request(request, "GET")
+        return super().check_permissions(request)
+
     @method_decorator(never_cache)
     def dispatch(self, *args, **kwargs):
         """Prevent caching when printing report templates"""
         return super().dispatch(*args, **kwargs)
 
+    def get_serializer(self, *args, **kwargs):
+        """Define a get_serializer method to be discoverable by the OPTIONS request."""
+        # Check the request to determine if the user has selected a label printing plugin
+        plugin = self.get_plugin(self.request)
+
+        serializer = plugin.get_printing_options_serializer(self.request)
+
+        # if no serializer is defined, return an empty serializer
+        if not serializer:
+            return serializers.Serializer()
+
+        return serializer
+
     def get(self, request, *args, **kwargs):
         """Perform a GET request against this endpoint to print labels"""
         common.models.InvenTreeUserSetting.set_setting('DEFAULT_' + self.ITEM_KEY.upper() + '_LABEL_TEMPLATE',
                                                        self.get_object().pk, None, user=request.user)
         return self.print(request, self.get_items())
 
+    def post(self, request, *args, **kwargs):
+        """Perform a GET request against this endpoint to print labels"""
+        common.models.InvenTreeUserSetting.set_setting('DEFAULT_' + self.ITEM_KEY.upper() + '_LABEL_TEMPLATE',
+                                                       self.get_object().pk, None, user=request.user)
+        return self.print(request, self.get_items())
```

Comment:
Following standards, a get method should not be used for mutating stuff and doing actions, which printing a label definitely is. Secondly a GET request doesn't have a body by spec. And with get, this would also get pretty complicated using the api forms.

Reply:
Oh, I got it, I guess you mean calling the get method internally in the post method.

Reply:
yes, that is what i meant